lib/ex_unit/lib/ex_unit/capture_log.ex

```diff
@@ -0,0 +1,84 @@
+defmodule ExUnit.CaptureLog do
+  @moduledoc ~S"""
+  Functionality to capture logs for testing.
+
+  ## Examples
+
+      defmodule AssertionTest do
+        use ExUnit.Case
+
+        import ExUnit.CaptureLog
+
+        test "example" do
+          assert capture_log(fn ->
+            Logger.error "log msg"
+          end) =~ "log msg"
+        end
+
+        test "check multiple captures concurrently" do
+          fun = fn ->
+            for msg <- ["hello", "hi"] do
+              assert capture_log(fn -> Logger.error msg end) =~ msg
+            end
+            Logger.debug "testing"
+          end
+          assert capture_log(fun) =~ "hello"
+          assert capture_log(fun) =~ "testing"
+        end
+      end
+
+  """
+
+  @doc """
+  Captures Logger messages generated when evaluating `fun`.
+
+  Returns the binary which is the captured output.
+
+  This function mutes the `:console` backend
+  and captures any log messages sent to Logger.
+
+  Note that when the `async` is set to `true`, the behaviour
+  is undetermined (e.g. the messages from another test might be captured).
+
+  It is possible to configure the level to capture with `:level`,
+  which will change the current Logger level for the duration of the
+  capture, for example, if the log level is changed to :error
+  any message with the lower level will be ignored.
+  The default level is `nil`, which will capture all messages.
```

Comment:
The Logger level is not changed currently, only the level that the backend will capture. I think if the `level` is set we should change it for the duration of the fun. If `level` is not `nil` and `async` is true the behaviour is undetermined because another process can change the level during a capture.